src/dmd/backend/dtype.d

```diff
@@ -582,22 +582,21 @@ type *type_delegate(type *tnext)
 
 /***********************************
  * Allocation a function type.
- * Input:
- *      tyf             function type
- *      ptypes[nparams] types of the function parameters
- *      variadic        if ... function
- *      tret            return type
+ * Params:
+ *      tyf      = function type
+ *      ptypes   = types of the function parameters
+ *      variadic = if ... function
+ *      tret     = return type
  * Returns:
  *      Tcount already incremented
  */
-extern (C) // because of size_t on OSX 32
-{
-type *type_function(tym_t tyf, type **ptypes, size_t nparams, bool variadic, type *tret)
+extern (C)
+type *type_function(tym_t tyf, type*[] ptypes, bool variadic, type *tret)
```

Comment:
Can we use `ref` instead of a pointer?

Reply:
No, because the pointer value escapes the function.

Reply:
We no longer support OSX32 so can we make this not `extern(C)`?